tests/system_tests_tcp_adaptor.py

```diff
@@ -1045,11 +1090,16 @@ def run_ncat(self,
                  timeout=10,
                  data=b'abcd',
                  use_ssl=False,
-                 use_client_cert=False):
+                 use_client_cert=False,
+                 use_bad_ca_cert=False,
```

Comment:
Just my opinion - you don't have to do it:

Why not just pass the actual client cert and key file as parameters rather than passing a bunch of flags and having run_ncat figure out what the certs are based on those flags?

IOW:

def run_ncat(... certfile=None, keyfile=None):...

Reply:
+1, I thought so too, but I did not want to suggest this as this would require rewriting the tests somewhat

also, `openssl s_client -connect` (https://www.openssl.org/docs/man1.0.2/man1/openssl-s_client.html) is maybe a more powerful tool for testing tls; but I see the point of just using ncat, since there are tests using ncat already (Openssl also have a server command, https://www.openssl.org/docs/man1.0.2/man1/s_server.html)